cmd/adsysd/main_test.go

```diff
@@ -114,3 +116,18 @@ func TestRun(t *testing.T) {
 		})
 	}
 }
+
+func TestMainApp(t *testing.T) {
+	if os.Getenv("ADSYS_CALL_MAIN") != "" {
+		main()
+		return
+	}
+
+	// #nosec G204: this is only for tests, under controlled args
+	cmd := exec.Command(os.Args[0], "version", "-test.run=TestMainApp")
```

Comment:
hum, how does it improve the coverage? We don’t build it and export GOCOVERDIR for this subprocess?

Reply:
ok, good for me then (I thought we still needed to build with -cover, but ok, trusting your results :))